### PR TITLE
79494 consume EES endpoint to publish event to EES

### PR DESCRIPTION
**Pull request by Nadeem14456**
PBI: 79494 contains below implementation

1. mock endpoint for EES in Dev Environment
2. ees endpoint to publish event
3. endpoint authentication
4. unit test cases

Please note that the erp-facade to EES connectivity is still in progress. We will make an attempt to publish the event on EES and share the results.

### CODE REVIEW

**Review by jrippington — APPROVED** [review]
Looks good

File: src/UKHO.ERPFacade.Common/Infrastructure/IEnumerableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace UKHO.ERPFacade.Common.Infrastructure
+{
+    [ExcludeFromCodeCoverage]
+    public static class IEnumerableExtensions
```
Comment (martyn-fewtrell) [inline src/UKHO.ERPFacade.Common/Infrastructure/IEnumerableExtensions.cs:6]: Is this correct?

Assume the name is misleading?

**Review by martyn-fewtrell — COMMENTED** [review]
Just one question/comment.

Otherwise looks good

**Review by harshal11869 — APPROVED** [review]
Looks Good !!!

File: src/UKHO.ERPFacade.Common/Infrastructure/IEnumerableExtensions.cs (hunk 1 above)
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.Common/Infrastructure/IEnumerableExtensions.cs]: yes. It does sound misleading. I renamed the IEnumerableExtensions class to IEnumerableComparerExtension.

File: src/UKHO.ERPFacade.Common/Infrastructure/IEnumerableComparerExtension.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace UKHO.ERPFacade.Common.Infrastructure
+{
+    [ExcludeFromCodeCoverage]
+    public static class IEnumerableComparerExtension
```
Comment (martyn-fewtrell) [inline src/UKHO.ERPFacade.Common/Infrastructure/IEnumerableComparerExtension.cs:6]: Thanks Nadeem

I assume this should be EnumerableComparerExtension as the "I" normally prefixes an Interface.

That being the case should this be excluded from Code Coverage?

File: src/UKHO.ERPFacade.Common/Infrastructure/EnumerableComparerExtension.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace UKHO.ERPFacade.Common.Infrastructure
+{
+    [ExcludeFromCodeCoverage]
```
[the remark this answers is not in the document]
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.Common/Infrastructure/EnumerableComparerExtension.cs]: The class checks for equivalence between two IEnumerable<T> objects. As a result, it was left out of the code coverage. We have added a unit test for it.

**Pull request merged**